store/sqlstore/channel_store.go

```diff
@@ -807,17 +807,31 @@ func (s SqlChannelStore) InvalidateChannelByName(teamId, name string) {
 }
 
 func (s SqlChannelStore) GetPinnedPosts(channelId string) (*model.PostList, error) {
-	pl := model.NewPostList()
+	builder := s.getQueryBuilder().Select(
+		"p.*",
+		"COALESCE(Threads.ReplyCount, 0) as ThreadReplyCount",
+		"COALESCE(Threads.LastReplyAt, 0) as LastReplyAt",
+		"COALESCE(Threads.Participants, '[]') as ThreadParticipants",
```

Comment:
> we're adding some additional fields as well

Yeah, since there is already a common method `PrepareThreadedResponse` which updates all these fields, so it's better to utilize it and populate all of them.

> Don't we also need IsFollowing from threadMemberships?

It's not exactly needed, and as it depends on userId which isn't available in this method. It seems unnecessary to update this method chain upto API layer, to include the userId.

Reply:
If the thread is known to the client it's OK to not have the `isFollowing` set since it will maintain the known value, but If it's not known yet would it be shown as un-followed?

Reply:
> but If it's not known yet would it be shown as un-followed?

Since it's an existing working code, I assume the current clients already have the necessary information and don't depend on this `isFollowing` flag. But yeah, in the future something new may come up that start using the `isFollowing` without knowing that it is never set.

Reply:
>Since it's an existing working code, I assume the current clients already have the necessary information and don't depend on this isFollowing flag

That's not necessarily a good assumption about our code 😛 

I checked in the webapp, pinned posts that I'm already following show up as unfollowed to me (it shows the "Follow" button, which means the app thinks I'm not following the thread)